<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from ./urdf/3segment_4chamber.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- This file has been generated automatically from create_xacro.cpp, do not edit by hand -->
<robot name="3segment_4chamber">
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <link name="base_link">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="0-base-ball-joint_x_joint" type="revolute">
    <parent link="base_link"/>
    <child link="0-base-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-base-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="0-base-ball-joint_y_joint" type="revolute">
    <parent link="0-base-ball-joint_link_0"/>
    <child link="0-base-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-base-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="0-base-ball-joint_z_joint" type="revolute">
    <parent link="0-base-ball-joint_link_1"/>
    <child link="0-a"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-a">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <!--creates a visual of a rod.-->
    <visual>
      <geometry>
        <cylinder length="0.055" radius="0.0022"/>
      </geometry>
      <material name="white"/>
      <origin xyz="0 0 0.0275"/>
    </visual>
  </link>
  <joint name="0-a-b_joint" type="prismatic">
    <parent link="0-a"/>
    <child link="0-b"/>
    <axis xyz="0 0 -1"/>
    <origin xyz="0 0 0.055"/>
    <limit effort="1000" lower="0" upper="0.055" velocity="1000"/>
  </joint>
  <link name="0-b">
    <visual>
      <geometry>
        <box size="0.0275 0.0275 0.0275"/>
      </geometry>
      <material name="red"/>
    </visual>
    <inertial>
      <mass value="0.12"/>
      <inertia ixx="0.0001" ixy="0.000" ixz="0.000" iyy="0.0001" iyz="0.000" izz="0.0001"/>
    </inertial>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="0-middle-ball-joint_x_joint" type="revolute">
    <parent link="0-b"/>
    <child link="0-middle-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-middle-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="0-middle-ball-joint_y_joint" type="revolute">
    <parent link="0-middle-ball-joint_link_0"/>
    <child link="0-middle-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-middle-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="0-middle-ball-joint_z_joint" type="revolute">
    <parent link="0-middle-ball-joint_link_1"/>
    <child link="0-c"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-c">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="0-c-d_joint" type="prismatic">
    <parent link="0-c"/>
    <child link="0-d"/>
    <axis xyz="0 0 -1"/>
    <origin xyz="0 0 0.055"/>
    <limit effort="1000" lower="0" upper="0.055" velocity="1000"/>
  </joint>
  <link name="0-d">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <!--creates a visual of a rod.-->
    <visual>
      <geometry>
        <cylinder length="0.055" radius="0.0022"/>
      </geometry>
      <material name="white"/>
      <origin xyz="0 0 -0.0275"/>
    </visual>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="0-tip-ball-joint_x_joint" type="revolute">
    <parent link="0-d"/>
    <child link="0-tip-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-tip-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="0-tip-ball-joint_y_joint" type="revolute">
    <parent link="0-tip-ball-joint_link_0"/>
    <child link="0-tip-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="0-tip-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="0-tip-ball-joint_z_joint" type="revolute">
    <parent link="0-tip-ball-joint_link_1"/>
    <child link="mid-0"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="mid-0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="1-base-ball-joint_x_joint" type="revolute">
    <parent link="mid-0"/>
    <child link="1-base-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-base-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="1-base-ball-joint_y_joint" type="revolute">
    <parent link="1-base-ball-joint_link_0"/>
    <child link="1-base-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-base-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="1-base-ball-joint_z_joint" type="revolute">
    <parent link="1-base-ball-joint_link_1"/>
    <child link="1-a"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-a">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <!--creates a visual of a rod.-->
    <visual>
      <geometry>
        <cylinder length="0.055" radius="0.0022"/>
      </geometry>
      <material name="white"/>
      <origin xyz="0 0 0.0275"/>
    </visual>
  </link>
  <joint name="1-a-b_joint" type="prismatic">
    <parent link="1-a"/>
    <child link="1-b"/>
    <axis xyz="0 0 -1"/>
    <origin xyz="0 0 0.055"/>
    <limit effort="1000" lower="0" upper="0.055" velocity="1000"/>
  </joint>
  <link name="1-b">
    <visual>
      <geometry>
        <box size="0.0275 0.0275 0.0275"/>
      </geometry>
      <material name="red"/>
    </visual>
    <inertial>
      <mass value="0.12"/>
      <inertia ixx="0.0001" ixy="0.000" ixz="0.000" iyy="0.0001" iyz="0.000" izz="0.0001"/>
    </inertial>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="1-middle-ball-joint_x_joint" type="revolute">
    <parent link="1-b"/>
    <child link="1-middle-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-middle-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="1-middle-ball-joint_y_joint" type="revolute">
    <parent link="1-middle-ball-joint_link_0"/>
    <child link="1-middle-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-middle-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="1-middle-ball-joint_z_joint" type="revolute">
    <parent link="1-middle-ball-joint_link_1"/>
    <child link="1-c"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-c">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="1-c-d_joint" type="prismatic">
    <parent link="1-c"/>
    <child link="1-d"/>
    <axis xyz="0 0 -1"/>
    <origin xyz="0 0 0.055"/>
    <limit effort="1000" lower="0" upper="0.055" velocity="1000"/>
  </joint>
  <link name="1-d">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <!--creates a visual of a rod.-->
    <visual>
      <geometry>
        <cylinder length="0.055" radius="0.0022"/>
      </geometry>
      <material name="white"/>
      <origin xyz="0 0 -0.0275"/>
    </visual>
  </link>
  <!--creates a ball joint, called from the PCC macro..-->
  <joint name="1-tip-ball-joint_x_joint" type="revolute">
    <parent link="1-d"/>
    <child link="1-tip-ball-joint_link_0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-tip-ball-joint_link_0">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
  <joint name="1-tip-ball-joint_y_joint" type="revolute">
    <parent link="1-tip-ball-joint_link_0"/>
    <child link="1-tip-ball-joint_link_1"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="1-tip-ball-joint_link_1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.011 0.011 0.011"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
  <joint name="1-tip-ball-joint_z_joint" type="revolute">
    <parent link="1-tip-ball-joint_link_1"/>
    <child link="mid-1"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000" lower="-0.785375" upper="0.785375" velocity="1000"/>
  </joint>
  <link name="mid-1">
    <!--since the RBDL needs a inertial info for each link, just assign a really small value to links where we don't have to consider inertia.-->
    <inertial>
      <mass value="0.00001"/>
      <inertia ixx="0.00001" ixy="0.0000" ixz="0.0000" iyy="0.00001" iyz="0.0000" izz="0.00001"/>
    </inertial>
  </link>
</robot>
